### PR TITLE
Add an MSBuild log to output on Windows

### PR DESCRIPTION
**Pull request by davkean**
This helps with build failure diagnostics. I've verified that this
had nearly-zero impact on build time.

### CODE REVIEW

**Comment by davkean** [comment]
Mac/Linux to come, trying to figure out the syntax.

tag @dotnet/roslyn-infrastructure @jaredpar

**Comment by jaredpar** [comment]
:+1: 

Have we changed Jenkins so this is saved as an artifact?  Essentially some way to click on this file directly vs. having to download the entire Binaries directory to see it.

**Comment by davkean** [comment]
@jaredpar How would I do that? I assumed that Jenkins just saved everything in binaries - suppose will find out when the build finishes.

**Comment by jaredpar** [comment]
@davkean it does.  But you have to download everything to see the file (giant ZIP).  At least that's the best UI experience that I know of.

File: cibuild.sh [hunk 1]
```diff
@@ -15,6 +15,7 @@ BUILD_CONFIGURATION=Debug
 OS_NAME=$(uname -s)
 USE_CACHE=true
 MONO_ARGS='--debug=mdb-optimizations --attach=disable'
+MSBUILD_ADDITIONALARGS = '/v:m  /consoleloggerparameters:Verbosity=minimal /filelogger /fileloggerparameters:Verbosity=normal'
```
Comment (jaredpar) [inline cibuild.sh]: No spaces around the = sign

File: cibuild.sh [hunk 2]
```diff
@@ -130,9 +131,9 @@ compile_toolset()
 {
     echo Compiling the toolset compilers
     echo -e "Compiling the C# compiler"
-    run_msbuild src/Compilers/CSharp/CscCore/CscCore.csproj /p:Configuration=$BUILD_CONFIGURATION
+    run_msbuild src/Compilers/CSharp/CscCore/CscCore.csproj /p:Configuration=$BUILD_CONFIGURATION /fileloggerparameters:LogFile=$(pwd)/Binaries/Bootstrap_CscCore.log
```
Comment (jaredpar) [inline cibuild.sh]: Do we need quotes around the path?

**Comment by amcasey** [comment]
@jaredpar I thought you could click on the zip link to see a web-based file explorer.

**Comment by jaredpar** [comment]
@davkean said the same.  Apparently I'm just terrible at using the Jenkins UI

**Comment by davkean** [comment]
Yep you can navigate straight to it: http://dotnet-ci.cloudapp.net/job/roslyn_prtest_lin_dbg_unit32/1254/artifact/Binaries/Build.log

**Comment by davkean** [comment]
Windows and Mac logs show up too: http://dotnet-ci.cloudapp.net/job/roslyn_prtest_win_dbg_unit64/1252/artifact/Binaries/Build.log and http://dotnet-ci.cloudapp.net/job/roslyn_prtest_mac_dbg_unit32/1228/artifact/Binaries/Build.log.

**Pull request merged**